### DISCUSSION TITLE
Tasmota ATHOM - Shutdown when the power falls below - Charging protection for pedelec batteries

### DISCUSSION THREAD

**Discussion by DaHuaba1** [arendst/Tasmota#17244] [Q&A]
Hello everyone,

I would like to program an ATHOM plug in such a way that it switches off when a PEDELEC battery has a charge capacity of around 80% and thus contributes to extending the battery life.

My idea would be that if the power consumption of the charger falls below a certain level (transition from CC to CV), the plug switches off.

The connector should be able to meet the following requirements:

1. After each plugging in, check which power is currently being used
2. At power >185 watts or from the beginning --> 'switch on'
3. If the power falls below <175 watts for 10 seconds --> 'switch off'

Since I usually rely on the full battery capacity for the commute, an additional rule would be necessary, which only charges the battery to >80% shortly before the trip:

4. Switching on again at 5:00 a.m. for a period of 30 minutes
5. Switching on again at 3:00 p.m. for a period of 30 minutes
6. this additional rule should only be activated as an option.

(Question: can the plug also be operated offline without WiFi or are there problems with the time? Unfortunately I don't have a WiFi connection in the office. If the Rule of Point #5 is not possible due to lack of WiFi than it's also o.k.)

Note: I am a ABSOLUTE novice when it comes to smart home applications or Tasmota programming, so answers and codes for laypeople would be very welcome.

Could you please help me to realize this project?
A complete Source Code would be very helpful :-) 

Currently I have temporarily programmed the following code:

code
Rule1 ON power1#state=1 DO Ruletimer1 10 ENDON ON Rules#Timer=1 DO Rule2 1 ENDON
Rule1 1
Rule2 ON Energy#Power<178 DO Backlog Power1 0; Rule2 0 ENDON

I would like to say thank you very much for your support.

**Comment by barbudor** [comment]
> Question: can the plug also be operated offline without WiFi or are there problems with the time
Tasmota is meant to run on Wifi
Without wifi, behavior is unpredictable especially anything related to time.


For the 1st part, the following should be enough:
```
Rule1
   on Energy#Power<178 do RuleTimer1 100 endon 
   on Energy#Power>178 do RuleTimer1 0 endon
Rule2 
   on Rules#Timer=1 do Power OFF endon
Rule1 1
Rule1 5
Rule2 1
```
Note that the rules are written here on multiple lines for readability but must be entered in a single line
It is important that Rule1 is set to Once mode (Rule1 5) so that the timer is starting once the Power cross the 178 value. Otherwise the timer will be restarted continuously and it won't do what you want.
The trigger on the timer is in another rule because it would have problem in a Rule1 because of the Once mode.

For the 2nd part, one way of doing is with a 3rd rule that you can enable whenever you want
```
Rule3 
   on Time#Minute=300 do backlog Rule2 0; Power ON endon
   on Time#Minute=330 do backlog Rule2 1; Power OFF endon
   on Time#Minute=900 do backlog Rule2 0; Power ON endon
   on Time#Minute=930 do backlog Rule2 1; Power OFF endon
```
You just have to enable Rule3 with `Rule3 1` to activate that option

Another possibility would be to use the Timers.
Enable timers of type "Rule":
- Timer 1 at 5:00
- Timer 2 at 5:30
- Timer 3 at 15:00
- Timer 4 at 15:30
and use Rule3 as:
```
Rule3 
   on Clock#Timer=1 do backlog Rule2 0; Power ON endon
   on Clock#Timer=2  do backlog Rule2 1; Power OFF endon
   on Clock#Timer=3 do backlog Rule2 0; Power ON endon
   on Clock#Timer=4  do backlog Rule2 1; Power OFF endon
Rule3 1
```
Leave Rule3 always enabled and control if you want the option by enable or not the timers
It is also easier to change the time at which you want to turn on/off using Timers

![image](https://user-images.githubusercontent.com/2996042/205365706-7d1bb8ed-705e-4d36-b419-d2e58fa12b84.png)

Note that the plug cannot detect when you plug the battery to charge as it is by default off
So once you have plugged the battery, you will have to press the button or send the Power ON from web/mqtt

**Reply by DaHuaba1** [reply]
Hello barbudor,

thank you very much for the code and detailled notes and instructions.
It took a little time but finally it works fine  😀.

Best regards,
Tom
